src/utils.jl

```diff
@@ -62,3 +62,50 @@ function vacuum_to_air(λ)
     n = 1 + 0.0000834254 + 0.02406147 / (130 - s^2) + 0.00015998 / (38.9 - s^2)
     λ / n
 end
+
+"""
+    parse_fwf(source, rowspec; datarow, lastrow)
+
+Parse fixed-width files, returning a vector of `NamedTuples`.
+`source` can be either an `AbstractVector` of `Strings` or a filename.
+`rowspec` should be a vector of tuples each containing, in order
+- a range object corresponding to the column postion
+- the type.  Strings will pull pulled directly other types will be parsed with `parse`.
+- the column name.
+- optionally, a function to apply to the parsed value.
+"""
+function parse_fwf(filename::String, rowspec; datarow=1, lastrow=0)
+    open(filename, "r") do file
+        for _ in 1:(datarow-1)
+            readline(file)
+        end
+        _parse_fwf_core(eachline(file), rowspec, lastrow-datarow+1)
+    end
+end
+function parse_fwf(lines::AbstractVector{String}, rowspec; datarow=1, lastrow=0)
+    lastrow = lastrow == 0 ? length(lines) : lastrow
+    _parse_fwf_core(lines[datarow:lastrow], rowspec)
+end
+
+function _parse_fwf_core(lines, rowspec, n=0)
+    #use n
+    map(lines) do line
+        (; _parse_item.(line, rowspec)...)
+    end
+end
+
+function _parse_item(line, itemspec)
+    if length(itemspec) == 4
+        r, t, n, f = itemspec
+    else
+        r, t, n = itemspec
+        f = identity
+    end
+    if t == String
+        (n, f(line[r]))
+    elseif r[2] > length(line) || strip(line[r]) == ""
```

Comment:
Minor question: why are you checking to see if the second index of the range is longer than the line length? Checking if the first entry of the range is greater than the line length or the last entry is grater than the line length both seem like more robust checks.